File: PascalTriangle.java
public class PascalTriangle {

    // Recursive method to get value at (row, col)
    public static int calculateValue(int row, int col) {
        if (col == 0 || col == row) return 1; // Base case
        return calculateValue(row - 1, col - 1) + calculateValue(row - 1, col); // Recursive step
    }

    // Method to print Triangle
    public static void printPascalTriangle(int numRows) {
        int maxWidth = String.valueOf(calculateValue(numRows - 1, numRows / 2)).length(); // Max number width
        for (int row = 0; row < numRows; row++) {
            // Print spaces
            for (int space = 0; space < (numRows - row - 1) * maxWidth; space++) {
                System.out.print(" ");
            }
            // Print current row
            for (int col = 0; col <= row; col++) {
                System.out.printf("%" + (maxWidth * 2) + "d", calculateValue(row, col));
            }
            System.out.println(); // Print the next row
        }
    }

    public static void main(String[] args) {
        int numRows = 17; //Change this value to print Pascal's Triangle with different number of rows
        System.out.println("Pascal's Triangle with " + numRows + " rows:");
        printPascalTriangle(numRows);
    }
}

File: PascalTriangle.py
def calculate_value(row, col):
    """
    Recursive method to get value at (row, col) in Pascal's Triangle.
    """
    if col == 0 or col == row:  # Base cases
        return 1
    return calculate_value(row - 1, col - 1) + calculate_value(row - 1, col)  # Recursive step


def print_pascal_triangle(num_rows):
    """
    Method to print Pascal's Triangle with the specified number of rows.
    """
    # Determine the maximum width of the largest number
    max_value = calculate_value(num_rows - 1, num_rows // 2)
    max_width = len(str(max_value))

    for row in range(num_rows):
        # Print leading spaces for alignment
        print(" " * ((num_rows - row - 1) * max_width), end="")

        # Print values in the current row with formatting
        for col in range(row + 1):
            print(f"{calculate_value(row, col):{max_width * 2}}", end="")

        print()  # Move to the next row


if __name__ == "__main__":
    num_rows = 13
    print(f"Pascal's Triangle with {num_rows} rows:")
    print_pascal_triangle(num_rows)
